tabs/src/components/BeaverHours/Scheduler.tsx

```diff
@@ -0,0 +1,158 @@
+import { createContext, useContext, useState } from "react";
+
+const ScheduleContext = createContext({
+  state: {} as OfficeHoursSchedule,
+  updateSchedule: (newSchedule: OfficeHoursSchedule) => {},
+});
+
+export function Scheduler() {
+  return <ScheduleForm />;
+}
+
+interface DaySchedule {
+  startAt?: string;
+  endAt?: string;
+  repeated?: number;
+}
+
+interface OfficeHoursSchedule {
+  [key: string]: DaySchedule | undefined;
+  Sunday?: DaySchedule;
+  Monday?: DaySchedule;
+  Tuesday?: DaySchedule;
+  Wednesday?: DaySchedule;
+  Thursday?: DaySchedule;
+  Friday?: DaySchedule;
+  Saturday?: DaySchedule;
+}
+
+export function ScheduleForm() {
+  const [state, setSchedule] = useState({
+    Sunday: {} as DaySchedule,
+    Monday: {} as DaySchedule,
+    Tuesday: {} as DaySchedule,
+    Wednesday: {} as DaySchedule,
+    Thursday: {} as DaySchedule,
+    Friday: {} as DaySchedule,
+    Saturday: {} as DaySchedule,
+  } as OfficeHoursSchedule);
+
+  const updateSchedule = (newSchedule: OfficeHoursSchedule) => {
+    setSchedule(newSchedule);
+  };
+  const value = { state, updateSchedule };
+
+  const HandleSubmit = (event: any) => {
+    console.log("A schedule was submitted:\n", state);
+    event.preventDefault();
+  };
+
+  return (
+    <ScheduleContext.Provider value={value}>
+      <form onSubmit={HandleSubmit}>
+        <GenerateWeekTable />
+        <input type="submit" value="submit" />
+      </form>
+    </ScheduleContext.Provider>
+  );
+}
+
+export function GenerateWeekTable() {
+  return (
+    <table>
+      <thead>
+        <tr>
+          <th></th>
+          <th>Start At:</th>
+          <th>End At:</th>
+          <th>Repeat for:</th>
+        </tr>
+      </thead>
+      <tbody>
+        {NewRow("Sunday")}
```

Comment:
If you do end up changing to the Syntax I suggested on line 84 then you can use it like 

`<CalendarSelectRow dayOfWeek={"Sunday"}/>`